pkg/controller/provider/container/ovirt/client.go

```diff
@@ -226,3 +234,365 @@ func GetInsecureSkipVerifyFlag(secret *core.Secret) bool {
 
 	return insecureSkipVerify
 }
+
+func DownloadOVFTestConection(secret *core.Secret) (ok bool, err error) {
+
+	var diskIdOVF string
+
+	url := fmt.Sprint(string(secret.Data["url"]), "/ovirt-engine/api")
+
+	// Create the connection to the server:
+	conn, err := ovirtsdk.NewConnectionBuilder().
+		URL(url).
+		Username(string(secret.Data["username"])).
+		Password(string(secret.Data["password"])).
+		Insecure(true).
+		Compress(true).
+		Timeout(time.Second * 10).
+		Build()
+	if err != nil {
+		err = liberr.Wrap(err)
+		return
+	}
+	defer conn.Close()
+
+	//Find OVF disk ID
+	disksService := conn.SystemService().DisksService()
+	diskResponse, err := disksService.List().Send()
+	if err != nil {
+		err = liberr.Wrap(err)
+		return
+	}
+
+	disks, ok := diskResponse.Disks()
+
+	if ok {
+		//get OVF disk ID
+		for _, disk := range disks.Slice() {
+			if diskName, ok := disk.Name(); ok {
+				if diskName == "OVF_STORE" {
+					if diskId, ok := disk.Id(); ok {
+						diskIdOVF = diskId
+					}
+					break
+				}
+			}
+		}
+	}
+
+	// Get a disk identified by uuid
+	diskService := disksService.DiskService(diskIdOVF)
+	diskRequest := diskService.Get()
+	diskRequestSent, err := diskRequest.Send()
+	if err != nil {
+		err = liberr.Wrap(err)
+		return
+	}
+	disk, ok := diskRequestSent.Disk()
+
+	diskId, ok := disk.Id()
+
+	//in case the OVF disk is locked wait fo OK status befor stating the image transfer
+	conn.WaitForDisk(diskId, ovirtsdk.DISKSTATUS_OK, 90*time.Second)
+
+	// Prepare image transfer request
+	transfersService := conn.SystemService().ImageTransfersService()
+	transfer := transfersService.Add()
+	image, err := ovirtsdk.NewImageBuilder().Id(diskId).Build()
+	if err != nil {
+		err = liberr.Wrap(err)
+		return
+	}
+	imageTransfer, err := ovirtsdk.NewImageTransferBuilder().Image(
+		image,
+	).Direction(
+		ovirtsdk.IMAGETRANSFERDIRECTION_DOWNLOAD,
+	).Build()
+	if err != nil {
+		err = liberr.Wrap(err)
+		return
+	}
+
+	// Initialize image transfer and lock the disk
+	transfer.ImageTransfer(imageTransfer)
+	transferReq, err := transfer.Send()
+	if err != nil {
+		err = liberr.Wrap(err)
+		return
+	}
+	currImageTransfer, ok := transferReq.ImageTransfer()
+
+	imageTransferId, ok := currImageTransfer.Id()
+
+	for {
+		if currImageTransfer.MustPhase() == ovirtsdk.IMAGETRANSFERPHASE_INITIALIZING {
+			time.Sleep(1 * time.Second)
+			currImageTransferReq, errIT := conn.SystemService().ImageTransfersService().ImageTransferService(imageTransferId).Get().Send()
+			if err != nil {
```

Comment:
you probably meant errIT here, but why having two different variables?

Reply:
for some reason the compiler having issue with using the original parameter so I had to create a new one.